code/src/beetle.cpp

```diff
@@ -28,47 +30,82 @@ namespace r2d2::moving_platform {
         // is made for -10 till 10.
         else if (speed < 10 && speed > -10) {
             speed = 0;
-            qik_2s12v10_motorcontroller.set_m0_speed(0);
-            qik_2s12v10_motorcontroller.set_m1_speed(0);
         }
+        qik_2s12v10_motorcontroller.set_m0_speed(speed);
+        qik_2s12v10_motorcontroller.set_m1_speed(-speed);
     }
 
     void beetle_c::set_steering(int16_t degrees) {
     }
     void beetle_c::turn(int16_t degrees) {
-        int new_degrees = degrees;
         // because of inaccuracies, when moving forward, the steer of manual
         // control will give a number beween -10 and 10 Just to make sure the
         // robot will move forward and not react to quickly, a theshold is made
         // for -15 till 15.
-        if (new_degrees < 15 && new_degrees > -15) {
-            new_degrees = 0;
+        int min_degrees = 15;
+        int max_degrees = 90;
+        if ((degrees < min_degrees && degrees > -min_degrees) ||
+            degrees > max_degrees || degrees < -max_degrees) {
+            degrees = 0;
         }
+        float turn = 2.40;
+        // The puls. Its starts low.
+        bool low_m0 = false;
+        int counter_m0 = 0;
+        bool low_m1 = false;
+        int counter_m1 = 0;
+        // The adc input. is between 3000 and 3800.
+        unsigned int adc_voltage = 3500;
+        // Encode frequency for 1 turn of the wheel. the encoder has 64 point
+        // per over 2 pins we count when de adc goes from low to high of 1 pin.
+        // The gear ratio from the motor is 50:1 64/4*50 = 800
+        int encode_1_full_turn = 800;
+        // Motor speed
```

Comment:
Comment unclear, What are the boundary value of motorspeed? if there are any at all, or is it a relative value?